Added new quickstart document for using embeds through new embed webapp (#13)
getsetup-io/embed-npm-package: docs/quickstart-upgrade.md

```html
<div id="iframe-target"></div>
```

```ts
import * as GSU from '@getsetup/embed'

// This function is supplied by the hosting page.
// This function will be called by the GSU iframe when it needs to navigate.
const navigationCallBack = (navigationAction: GSU.NavigationAction, classId: string) => {
  let href = `/hostSitePath/${navigationAction}`

  // The 'watch' navigation action is special because we need to pass the classId to the 'watch' page.
  if (navigationAction === 'watch') {
    href = href + `?classId=${classId}`
  }
  window.location.href = href
}

// Token exchange is not yet implemented with the embed webapp and not required in case of most of the partners
const tokenRequestCallBack = () => {}

GSU.createIframe({
  targetElementId: 'iframe-target',
  targetPage: 'discover',
  partnerId: 'gsudemo',
  navigationCallBack,
  tokenRequestCallBack,
  // This targetUrls object isn't necessary in production integrations, but it allows us to point to the dev environment.
  targetUrls: {
    discovery: 'https://embed-webapp.www.gsudevelopment.com/discovery/{partnerId}',
    watch: 'https://embed-webapp.www.gsudevelopment.com/watch/{partnerId}/{classId}',
  },

> })

Reviewer: We should fix it in the package, and make it `discovery` if that's what we have in the URL.

// This function is supplied by the hosting page.
// This function will be called by the GSU iframe when it needs to navigate.
const navigationCallBack = (navigationAction: GSU.NavigationAction, classId: string) => {
  let href = `/hostSitePath/${navigationAction}`

  // The 'watch' navigation action is special because we need to pass the classId to the 'watch' page.
  if (navigationAction === 'watch') {
    href = href + `?classId=${classId}`
  }
  window.location.href = href
}

// Token exchange is not yet implemented with the embed webapp and not required in case of most of the partners
const tokenRequestCallBack = () => {}

//Get the sessionId from the query string.
const urlParams = new URLSearchParams(window.location.search)
const classId = urlParams.get('classId')

GSU.createIframe({
  targetElementId: 'iframe-target',
  targetPage: 'watch',
  classId: classId,
  partnerId: 'gsudemo',
  navigationCallBack,
  tokenRequestCallBack,
  // This targetUrls object isn't necessary in production integrations, but it allows us to point to the dev environment.
  targetUrls: {
    discovery: 'https://embed-webapp.www.gsudevelopment.com/discovery/{partnerId}',

>     watch: 'https://embed-webapp.www.gsudevelopment.com/watch/{partnerId}/{classId}',

Reviewer: `discover`

# Setup

## CORS & CSP

> The GetSetUp Embedded product requires some CORS and CSP configuration by GetSetUp to allow other sites access. We have configured access for http://localhost:3001. If you require access for other development urls please contact deepak@getsetup.io.

Reviewer: engineering@getsetup.io

## Site layout

For the example below we will assume a two page site. The first page will host the "discovery" page that embeds a list of classes offered by GetSetUp.
The second page is the "watch" page that hosts the video viewing and chat experience.

## How to install locally

In your project, run `npm install @getsetup/embed`
The next step is to import and configure the package on a couple of pages.

### Discovery page

First on a page where you want the class browsing experience create a div and setup the iframe:

```html
<div id="iframe-target"></div>
```

```ts
import * as GSU from '@getsetup/embed'

// This function is supplied by the hosting page.
// This function will be called by the GSU iframe when it needs to navigate.

> const navigationCallBack = (

Reviewer: Should specify mentioning the `partnerId` also.

  // The 'watch' navigation action is special because we need to pass the classId to the 'watch' page.
  if (navigationAction == "watch" && classId) {
    targetUrl.pathname = "/hostSitePath/watch";
    targetUrl.searchParams.set("class-id", classId ?? "");
  } else if (navigationAction == "discover") {
    targetUrl.pathname = "/hostSitePath/discover";
    targetUrl.searchParams.set("partner-id", partnerId);
    targetUrl.searchParams.delete("class-id");
  }
  window.location.assign(targetUrl);
};

// Token exchange is not yet implemented with the embed webapp and not required in case of most of the partners
const tokenRequestCallBack = () => {}

//Get the sessionId from the query string.
const urlParams = new URLSearchParams(window.location.search)
const classId = urlParams.get('classId')

GSU.createIframe({
  targetElementId: 'iframe-target',
  targetPage: 'watch',
  classId: classId,
  partnerId: 'gsudemo',
  navigationCallBack,
  tokenRequestCallBack,
  // This targetUrls object isn't necessary in production integrations, but it allows us to point to the dev environment.
  targetUrls: {
    discover: 'https://embed-webapp.www.gsudevelopment.com/discovery/{partnerId}',

>     watch: 'https://embed-webapp.www.gsudevelopment.com/watch/{partnerId}/{classId}',

Reviewer: These have to be mentioned separately in the docs too, 
prod URLs and dev URLs.

Reviewer: Note: These are overridden URLs, used and specified only while testing locally or in a staging environment. 
We can skip these in the production, as the embed package by default points to the production ones.

  },
})
```

## Use the site

> Now that both pages are setup, you can go to the page hosting the "learn" page and click on a class. The iframe will request that the hosting page navigates to the "joinClass" page using the `navigationCallBack` we setup.

Reviewer: This will need updates.

// Read partner id from query string or hardode it
const searchParams = new URLSearchParams() // Please feel free to use framework specific equivalent functions here for e.g. React has useSearchParams().
const partnerId = searchParams.get('partner-id') || 'OD3rF3b8P'

// This function is supplied by the hosting page.
// This function will be called by the GSU iframe when it needs to navigate.
const navigationCallBack = ({
  navigationAction,
  classId,
}: {
  navigationAction: GSU.NavigationAction
  classId?: string
}) => {
  const targetUrl = new URL(window.location.href)
  // The 'watch' navigation action is special because we need to pass the classId to the 'watch' page.
  if (navigationAction == 'watch' && classId) {
    targetUrl.pathname = '/hostSitePath/watch'
    targetUrl.searchParams.set('class-id', classId ?? '')
  } else if (navigationAction == 'discover') {
    targetUrl.pathname = '/hostSitePath/discover'
    targetUrl.searchParams.set('partner-id', partnerId)
    targetUrl.searchParams.delete('class-id')
  }
  window.location.assign(targetUrl)
}

// Token exchange is not yet implemented with the embed webapp and not required in case of most of the partners

> const tokenRequestCallBack = () => {}

Reviewer: ```suggestion
// Token exchange is handled on a case-by-case basis. Pass this if it's not to be implemented. 
const tokenRequestCallBack = () => {}
```

```

```ts
import * as GSU from '@getsetup/embed'

// Read class id and partner id from query string or hardode it
const searchParams = new URLSearchParams() // Please feel free to use framework specific equivalent functions here for e.g. React has useSearchParams().
const classId = searchParams.get('class-id') || ''
const partnerId = searchParams.get('partner-id') || 'OD3rF3b8P'

// This function is supplied by the hosting page.
// This function will be called by the GSU iframe when it needs to navigate.
const navigationCallBack = ({
  navigationAction,
  classId,
}: {
  navigationAction: GSU.NavigationAction
  classId?: string
}) => {
  const targetUrl = new URL(window.location.href)
  // The 'watch' navigation action is special because we need to pass the classId to the 'watch' page.
  if (navigationAction == 'watch' && classId) {
    targetUrl.pathname = '/hostSitePath/watch'
    targetUrl.searchParams.set('class-id', classId ?? '')
  } else if (navigationAction == 'discover') {
    targetUrl.pathname = '/hostSitePath/discover'
    targetUrl.searchParams.set('partner-id', partnerId)
    targetUrl.searchParams.delete('class-id')
  }
  window.location.assign(targetUrl)
}

> // Token exchange is not yet implemented with the embed webapp and not required in case of most of the partners
> const tokenRequestCallBack = () => {}

Reviewer: ```suggestion
// Token exchange is handled on a case-by-case basis. Pass this if it's not to be implemented. 
const tokenRequestCallBack = () => {}
```
